src/main/java/uk/gov/ons/ssdc/supporttool/endpoint/EmailTemplateEndpoint.java

```diff
@@ -77,4 +72,23 @@ public EmailTemplateEndpoint(
 
     return new ResponseEntity<>(HttpStatus.CREATED);
   }
+
+  private void validateEmailTemplate(EmailTemplateDto emailTemplateDto) {
+
+    emailTemplateRepository
+        .findAll()
+        .forEach(
+            emailTemplate -> {
+              if (emailTemplate.getPackCode().equalsIgnoreCase(emailTemplateDto.getPackCode())) {
+                throw new ResponseStatusException(
+                    HttpStatus.BAD_REQUEST, "Pack code already exists");
+              }
+            });
+
+    Set<String> templateSet = new HashSet<>(Arrays.asList(emailTemplateDto.getTemplate()));
```

Comment:
Would it be a good idea to check for duplicates first, then check for an existing pack code after this, as it's using the database?